### PR TITLE
fix(ui): improve identity and project name validation

### LINKED ISSUE #949: Make user handle and project name validation messages friendlier
[While doing QA](https://github.com/radicle-dev/radicle-upstream/issues/912#issuecomment-693499113.) we discovered that the display name and project name validation messages can be confusing to users. Particularly it's not instantly obvious why a single character is not valid (names have to be at least 2 characters).

Currently we only show the regex that the input is validated against.

Screenshot 2020-09-22 at 15 38 01
Screenshot 2020-09-22 at 15 38 14

My proposal is to split up the validations in parts and write a separate message for each of them, like so:

- **display name:**
  - Must start with a letter or digit
  - Must be at least 2 characters long
  - Only letters `a`-`z`, `A`-`Z`, digits `0`-`9` and special characters `-`, `_` are allowed

- **project name:**
  - Must start with a letter or digit
  - Must be at least 2 characters long
  - Only letters `a`-`z`, `A`-`Z`, digits `0`-`9` and special characters `-`, `_`, `.` are allowed

We would still want to show only one validation at a time to save space.

### PR DESCRIPTION
**Pull request by MeBrei**
Closes #949

### CODE REVIEW

File: ui/src/validation.ts [hunk 1]
```diff
@@ -48,8 +52,22 @@ interface FormatConstraints {
...
+validatejs.validators.firstHandleChar = (
+  value: string,
+  options: firstHandleCharOptions,
+  key: any,
+  attributes: any
+) => {
+  if (!value.match(new RegExp("^[a-z0-9]", "i"))) {
+    return (
+      `Your ${  options.valueName  } should start with a letter or a number.`
+    );
```
Comment (rudolfs) [inline ui/src/validation.ts:67]:
```suggestion
    return `Your ${options.valueName} should start with a letter or a number.`;
```

**Review by rudolfs — APPROVED** [review]

**Review by rudolfs — DISMISSED** [review]

File: ui/src/validation.ts [hunk 2]
```diff
@@ -39,6 +39,10 @@ interface RemoteValidation {
   validationMessage: string;
 }
 
+interface firstHandleCharOptions {
```
Comment (sarahscott) [inline ui/src/validation.ts:42]:
```suggestion
interface FirstHandleCharOptions {
```

File: ui/src/validation.ts [hunk 3]
```diff
@@ -48,8 +52,20 @@ interface FormatConstraints {
     pattern: RegExp;
     message: string;
   };
+  firstHandleChar?: firstHandleCharOptions;
```
Comment (sarahscott) [inline ui/src/validation.ts:55]: Alternatively, you could skip making a separate `interface` and just do:
```suggestion
  firstHandleChar?: {
      valueName: string
  };
```

**Review by sarahscott — CHANGES_REQUESTED** [review]
Looking good apart from the `interface` thing - should be capitalized or built into the existing `FormatConstraints` ⭐

**Review by rudolfs — APPROVED** [review]

**Review by sarahscott — APPROVED** [review]

**Pull request merged**